### PR TITLE
Adapt to use PyPi for builds instead of RPMs

### PR DESCRIPTION
**Pull request by dulek**
Main problem with RPMs is that we need to use OSP ones and these are
super outdated. ART recently figured out how to enable builds based on
PyPi packages going through ARTs Cachito instance. This will let us skip
all the RPM hell that we have today and this commit aims to do that. In
particular:

1. kuryr-requirements.txt file is added that will gather exact versions
   of packages that we want from PyPi.
2. Dockerfiles are adapted to install Kuryr as a Python package and not
   an RPM.
3. Stuff required to build Kuryr RPM is removed as it's no needed
   anymore.

### CODE REVIEW

**Comment by dulek** [comment]
/hold

This is strongly WiP!

**Comment by dulek** [comment]
/retest

That's just a failure to acquire the lease.

File: setup.cfg [hunk 1]
```diff
@@ -1,5 +1,7 @@
 [metadata]
 name = kuryr-kubernetes
+# Needs to be higher than tag for pbr to stop complaining
+version = 99.99.99
```
Comment (dtantsur) [inline setup.cfg:4]: Alternative: https://github.com/openshift/ironic-agent-image/blob/main/Dockerfile.ocp#L16

**Comment by dulek** [comment]
We might want to revisit this at some point, but right now it's not a priority.

**Pull request closed without merging**